### PR TITLE
[mle] validate CSL Channel TLV in `HandleChildUpdateRequestOnParent()`

### PR DESCRIPTION
**Pull request by soma0212**
The CSL Channel TLV in a Child Update Request is applied to the child as-is:
- `HandleChildUpdateRequestOnParent()` truncates the 16-bit channel to `uint8_t` and stores it with no range or channel-page check, unlike the local `otLinkSetCslChannel()` path which goes through `Radio::IsCslChannelValid()`
- `CslTxScheduler::HandleFrameRequest()` uses the stored value for `frame->SetChannel()`, so it reaches `otPlatRadioTransmit()` / `otPlatRadioReceive()` unchanged and an attached child can make the parent hand the radio an arbitrary channel (the simulation radio indexes `sChannelMaxTransmitPower[channel - kMinChannel]` by it)
- values such as `0x011A` silently become channel 26 after the truncation
The request is now rejected as a parse error unless the channel is zero (unspecified) or passes `ChannelTlvValue::IsValid()`. A unit test in `test_mle.cpp` drives the handler with crafted requests.

### CODE REVIEW

File: src/core/thread/mle_ftd.cpp [hunk 1]
```diff
@@ -2360,9 +2360,12 @@ void Mle::HandleChildUpdateRequestOnParent(RxInfo &aRxInfo)
 
         if (Tlv::Find<CslChannelTlv>(aRxInfo.mMessage, cslChannelTlvValue) == kErrorNone)
         {
+            uint16_t cslChannel = cslChannelTlvValue.GetChannel();
+
             // Special value of zero is used to indicate that
             // CSL channel is not specified.
-            child->SetCslChannel(static_cast<uint8_t>(cslChannelTlvValue.GetChannel()));
+            VerifyOrExit((cslChannel == 0) || cslChannelTlvValue.IsValid(), error = kErrorParse);
+            child->SetCslChannel(static_cast<uint8_t>(cslChannel));
```
[the remark this answers is not in the document]
Comment (soma0212) [reply src/core/thread/mle_ftd.cpp]: Done, dropped the local and call GetChannel() directly in both spots.

File: src/core/thread/mle_ftd.cpp [hunk 2]
```diff
@@ -2362,6 +2362,7 @@ void Mle::HandleChildUpdateRequestOnParent(RxInfo &aRxInfo)
         {
             // Special value of zero is used to indicate that
             // CSL channel is not specified.
+            VerifyOrExit((cslChannelTlvValue.GetChannel() == 0) || cslChannelTlvValue.IsValid(), error = kErrorParse);
```
Comment (abtink) [inline src/core/thread/mle_ftd.cpp]: @soma0212, thanks for finding and fixing this. The current code is correct, but I have a suggestion that might be a cleaner approach:

We can declare a new class in `mle_tlvs.hpp` for `CslChannelTlvValue` that inherits from `ChannelTlvValue` and overrides `IsValid()`:

```cpp
OT_TOOL_PACKED_BEGIN
class CslChannelTlvValue : public ChannelTlvValue
{
public:
    /**
     * Indicates whether or not the CSL Channel and Channel Page values are valid.
     *
     * @retval TRUE   If the Channel and Channel Page values are valid.
     * @retval FALSE  If the Channel and Channel Page values are not valid.
     */
    bool IsValid(void) const;
} OT_TOOL_PACKED_END;

```

We can make `IsValid()` for this type allow zero in addition to what the base class accepts:

```cpp
bool CslChannelTlvValue::IsValid(void) const
{
    // Special value of zero is used to indicate that
    // CSL channel is not specified.

    return (GetChannel() == 0) || ChannelTlvValue::IsValid();
}

```

And then use it for the `CslChannelTlv` definition:

```cpp
/**
 * Defines CSL Channel TLV constants and types.
 */
typedef SimpleTlvInfo<Tlv::kCslChannel, CslChannelTlvValue> CslChannelTlv;

```

This way, the TLV and its associated value type encapsulate the check allowing zero for `IsValid()`, ensuring it is handled consistently everywhere it's used.

Comment (soma0212) [reply src/core/thread/mle_ftd.cpp]: Makes sense, encapsulating it in the value type is cleaner than special-casing zero at the call site. Pushed it: `CslChannelTlvValue` inherits from `ChannelTlvValue` and overrides `IsValid()` to accept zero, and `CslChannelTlv` now uses it, so the handler just checks `IsValid()`. I gave it constructors mirroring the base since `AppendCslChannelTlv()` and the unit test construct the value directly. Full unit suite still passes.

**Review by abtink — APPROVED** [review]
@soma0212 , looks great. Thanks for changes. 👍

File: tests/unit/test_mle.cpp [hunk 3]
```diff
@@ -503,6 +503,99 @@ class UnitTester
...
+    {
+        static constexpr uint16_t kCslPeriod = 3125;
+
+        struct TestCase
+        {
+            uint16_t mChannel;
+            bool     mShouldAccept;
+        };
+
+        static const TestCase kTestCases[] = {
```
Comment (jwhui) [inline tests/unit/test_mle.cpp:518]: Consider adding a test case for a non-zero channel value below `Radio::kChannelMin` to verify that non-zero channels below the minimum bound are rejected while 0 continues to be accepted.

File: tests/unit/test_mle.cpp [hunk 4]
```diff
@@ -503,6 +503,99 @@ class UnitTester
         testFreeInstance(instance);
         printf("TestTxChallengeTable passed\n");
     }
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    static void HandleChildUpdateRequest(Mle::Mle &aMle, Message &aMessage, const Mac::ExtAddress &aChildExtAddress)
```
Comment (jwhui) [inline tests/unit/test_mle.cpp]: Move to `private:` section?

File: tests/unit/test_mle.cpp (hunk 3 above)
Comment (soma0212) [reply tests/unit/test_mle.cpp:518]: Added `{Radio::kChannelMin - 1, false}` to the cases; 0 is still covered as accepted right above it. Test passes.

File: tests/unit/test_mle.cpp (hunk 4 above)
Comment (soma0212) [reply tests/unit/test_mle.cpp]: Done, moved it to the `private:` section, guarded with `OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE` since that section sits outside the FTD block.

**Comment by jwhui** [comment]
@soma0212 , please rebase and resolve conflicts.

**Comment by soma0212** [comment]
Rebased on main. The only conflict was in test_mle.cpp where TestRouterTableRouterIdBounds() landed at the same spot in main(); kept both test invocations. ot-test-mle passes after the rebase.

File: src/core/thread/mle_ftd.cpp [hunk 5]
```diff
@@ -2360,8 +2360,7 @@ void Mle::HandleChildUpdateRequestOnParent(RxInfo &aRxInfo)
 
         if (Tlv::Find<CslChannelTlv>(aRxInfo.mMessage, cslChannelTlvValue) == kErrorNone)
```
Comment (jwhui) [inline src/core/thread/mle_ftd.cpp]: If `Tlv::Find<CslChannelTlv>` encounters a malformed TLV (e.g. invalid TLV length), it returns `kErrorParse`. With `if (Tlv::Find(...) == kErrorNone)`, `kErrorParse` is silently ignored and treated as if the TLV was not found.

To be consistent with how other TLVs are parsed in this function (such as `TimeoutTlv` and `SupervisionIntervalTlv`), consider switching on the return value.

Comment (soma0212) [reply src/core/thread/mle_ftd.cpp]: Good point, a bad length would fall through as if the TLV were absent. Switched it to the same switch pattern as the Timeout and Supervision Interval TLVs, so kErrorParse from Find now rejects the request. ot-test-mle still passes.

**Comment by jwhui** [comment]
@soma0212 , can you help squash commits and update commit message?

**Comment by soma0212** [comment]
Squashed into a single commit and rewrote the message to describe the final shape of the change: the `CslChannelTlvValue` type, the switch on the `Tlv::Find` result, and the unit test. Rebased onto current main at the same time since the branch had fallen behind; no conflicts, the resulting diff is unchanged, and ot-test-mle passes.

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Comment by soma0212** [comment]
Thanks for landing this. The CslChannelTlvValue shape ended up cleaner than the original call-site check.
